File: JavaScript/Hashmap & Set/2352. Equal Row and Column Pairs.js
/* 2352. Equal Row and Column Pairs
Difficulty: Medium
Submission Link: https://leetcode.com/problems/equal-row-and-column-pairs/submissions/1385750832/
*/

var equalPairs = function (grid) {
    let res = 0
    for (let i = 0; i < grid.length; ++i) {
        let row = grid[i].join()
        for (let j = 0; j < grid.length; ++j) {
            let col = grid.map((v) => v[j]).join()
            if (col == row) {
                res++
            }
        }
    }
    return res
};

File: C/Hashmap & Set/2352. Equal Row and Column Pairs.c
/* 2352. Equal Row and Column Pairs
Difficulty: Medium
Submission Link: https://leetcode.com/problems/equal-row-and-column-pairs/submissions/1385753084/ */

int equalPairs(int** grid, int gridSize, int* gridColSize){
    int ansCnt = 0;
    int rLen = gridSize, cLen = *gridColSize;
    for(int i = 0; i < rLen; i++){
        for(int j = 0; j < cLen; j++){
            if(grid[i][0] == grid[0][j]){
                ansCnt++;
                for(int k = 0; k < rLen; k++)
                {
                    if(grid[i][k]!=grid[k][j])
                    {
                        ansCnt--;
                        break;
                    }
                }
            }
        }
    }
    return ansCnt;
}